src/controllers/sold.land.filter.controller.ts

```diff
@@ -28,18 +33,39 @@ export const get = (req: Request, res: Response, next: NextFunction) => {
   }
 };
 
-export const post = (req: Request, res: Response, next: NextFunction) => {
+export const post = async (req: Request, res: Response, next: NextFunction) => {
   try {
+
     logger.debugRequest(req, `POST ${config.SOLD_LAND_FILTER_PAGE}`);
+
+    const session = req.session as Session;
     const hasSoldLand = req.body[HasSoldLandKey];
+    const appData: ApplicationData = await getApplicationData(session, req);
+    appData[HasSoldLandKey] = hasSoldLand;
+
+    let nextPageUrl: string = "";
 
-    setExtraData(req.session, { ...getApplicationData(req.session), [HasSoldLandKey]: hasSoldLand });
+    if (hasSoldLand !== '0') {
+      nextPageUrl = config.CANNOT_USE_URL;
+    } else {
+      if (!appData[Transactionkey]) {
```

Comment:
I was expecting this change in behaviour to be feature flagged but it's not. That _might_ be ok, but we need to discuss. Relates to this comment on the API change:

https://github.com/companieshouse/overseas-entities-api/pull/404/files#r1673848116

Reply:
To me, it makes sense to put this behind the Redis feature flag as it's a fundamental change to the behaviour.

Reply:
Agreed. Feature flag added to new PR [here](https://github.com/companieshouse/overseas-entities-web/pull/1471).